tests/Doctrine/Tests/ORM/Functional/SelfReferencingTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\ORM\Annotation as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class SelfReferencingTest extends OrmFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+        try {
+            $classMetadatas = [
+                $this->em->getClassMetadata(Issue7877ApplicationGenerated::class),
+                $this->em->getClassMetadata(Issue7877DatabaseGenerated::class),
+            ];
+            // We first drop the schema to avoid collision between tests
+            $this->schemaTool->dropSchema($classMetadatas);
+            $this->schemaTool->createSchema($classMetadatas);
+        } catch (Exception $e) {
+        }
+    }
+
+    public function providerDifferentEntity()
+    {
+        yield [Issue7877ApplicationGenerated::class];
+        yield [Issue7877DatabaseGenerated::class];
+    }
+
+    /**
+     * @dataProvider providerDifferentEntity
+     */
+    public function testDifferentEntity(string $class)
```

Comment:
What's being tested specifically? Can you clarify that via method names? (applies to all methods in this test)

Reply:
fixed